brion/compartmentReport.h

```diff
@@ -115,41 +118,45 @@ class CompartmentReport : public boost::noncopyable
      *
      * @param index neuron index per current GID set
      * @return number of compartments for the given neuron
-     * @version 1.0
+     * @version 2.0
      */
     BRION_API size_t getNumCompartments(size_t index) const;
 
-    /** @return the current start time of the report. @version 1.0 */
+    /** @return the current start time of the report. @version 2.0 */
     BRION_API float getStartTime() const;
 
-    /** @return the current end time of the report. @version 1.0 */
+    /** @return the current end time of the report. @version 2.0 */
     BRION_API float getEndTime() const;
 
-    /** @return the sampling time interval of the report. @version 1.0 */
+    /** @return the sampling time interval of the report. @version 2.0 */
     BRION_API float getTimestep() const;
 
-    /** @return the data unit of the report. @version 1.0 */
+    /** @return the data unit of the report. @version 2.0 */
     BRION_API const std::string& getDataUnit() const;
 
-    /** @return the time unit of the report. @version 1.0 */
+    /** @return the time unit of the report. @version 2.0 */
     BRION_API const std::string& getTimeUnit() const;
 
-    /** @return the number of values of a loaded report frame. @version 1.0 */
+    /** @return the number of values of a loaded report frame. @version 2.0 */
     BRION_API size_t getFrameSize() const;
 
     /** Load report values at the given time stamp.
      *
      * @param timestamp the time stamp of interest
      * @return the report values if found at timestamp, nullptr otherwise
-     * @version 1.0
+     * @note Until the completion of this operation, the report shouldn't be
+     * modified
+     * @version 2.0
      */
-    BRION_API floatsPtr loadFrame(float timestamp) const;
+    BRION_API std::future<floatsPtr> loadFrame(float timestamp) const;
+
+    BRION_API std::future<Frames> loadFrames(float start, float end) const;
```

Comment:
Add documentation. For example, it's not obvious that if the requested window is not contained in the report window, this functions returns an empty frame list instead.
On the other hand, I'm not sure that I agree with that behaviour. If you don't change the functions to throw instead of returning an empty array, I think that loadFrames should load as many frames as possible within the start, end input interval (regardless of it being larger the report time window).